### PR TITLE
🌐 [i18n-KO] Translated `optimizers.md` to Korean

### PR DESCRIPTION
**Pull request by chelsseeey**
# What does this PR do?

Translated the `optimizers.md` file of the documentation to Korean.
Thank you in advance for your review.

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [X] Check for missing / redundant translations (번역 누락/중복 검사)
- [X] Grammar Check (맞춤법 검사)
- [X] Review or Add new terms to glossary (용어 확인 및 추가)
- [X] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [X] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

May you please review this PR?
@jungnerd, @luckyvickyricky, @chelsseeey, @skwh54, @amo33, @maximizemaxwell, @D15M4S 

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/_toctree.yml [hunk 1]
```diff
@@ -295,86 +295,6 @@
...
-    - local: contributing
-      title: 🤗 Transformers에 기여하는 방법
-    - local: add_new_model
-      title: 🤗 Transformers에 새로운 모델을 추가하는 방법
-    - local: add_new_pipeline
-      title: 어떻게 🤗 Transformers에 파이프라인을 추가하나요?
-    - local: testing
-      title: 테스트
-    - local: pr_checks
-      title: Pull Request에 대한 검사
```
Comment (nevertmr) [inline docs/source/ko/_toctree.yml]: 이부분 삭제 되어있네요!

File: docs/source/ko/optimizers.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+)
+```
+
+## Schedule Free[[schedule-free]]
+
+```bash
+pip install schedulefree
+```
+
+[Schedule Free optimizer (SFO)](https://hf.co/papers/2405.15682)는 기본 옵티마이저의 모멘텀 대신 평균화(averaging)와 보간(interpolation)을 조합하여 사용합니다. 덕분에 기존의 학습률 스케줄러처와 달리, SFO는 학습률을 점진적으로 낮추는 절차가 아예 필요 없습니다.
```
Comment (nevertmr) [inline docs/source/ko/optimizers.md]:
```suggestion
[Schedule Free optimizer (SFO)](https://hf.co/papers/2405.15682)는 기본 옵티마이저의 모멘텀 대신 평균화(averaging)와 보간(interpolation)을 조합하여 사용합니다. 덕분에 기존의 학습률 스케줄러와 달리, SFO는 학습률을 점진적으로 낮추는 절차가 아예 필요 없습니다.
```

File: docs/source/ko/optimizers.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+> [!TIP]
+> `scale` 파라미터는 `n/r`으로 설정할 수 있습니다. 이때, `n`은 원본 공간 차원이고 `r`은 저랭크(low-rank) 공간 차원입니다. `scale`을 기본값으로 유지하면서 학습률만 조정해도 비슷한 효과를 얻을 수 있습니다.
+
+| parameter | description | APOLLO | APOLLO-Mini |
+|---|---|---|---|
+| rank | rank of the auxiliary sub-space for gradient scaling | 256 | 1 |
+| scale_type | how scaling factors are applied | `channel` (per-channel scaling) | `tensor` (per-tensor scaling) |
+| scale | adjusts gradient updates to stabilize training | 1.0 | 128 |
+| update_proj_gap | steps before updating projection matrices | 200 | 200 |
+| proj | projection type | `random` | `random` |
```
Comment (nevertmr) [inline docs/source/ko/optimizers.md]: 일반적으로 표내용도 번역하는 것 같아요!

https://github.com/huggingface/transformers/blob/738c1a3899d361274a02f2306f46e70fd8c2bd46/docs/source/ko/cache_explanation.md?plain=1#L59-L62

File: docs/source/ko/optimizers.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+```diff
+args = TrainingArguments(
+    output_dir="./test-lomo",
+    max_steps=1000,
+    per_device_train_batch_size=4,
++   optim="adalomo",
+    gradient_checkpointing=True,
+    gradient_checkpointing=True,
+    logging_strategy="steps",
```
Comment (nevertmr) [inline docs/source/ko/optimizers.md]:
```suggestion
+   optim="adalomo",
    gradient_checkpointing=True,
    logging_strategy="steps",
```

원문에도 중복이 있어서 #40014 PR날렸고, 번역문에서도 한줄 삭제해주시면 될 것 같습니다!̤̻

File: docs/source/ko/optimizers.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+)
+```
+
+## APOLLO[[apollo]]
+
+```bash
+pip install apollo-torch
+```
+
+[Approximated Gradient Scaling for Memory Efficient LLM Optimization (APOLLO)](https://github.com/zhuhanqing/APOLLO) 는 사전 학습과 파인튜닝 모두에 대해 전체 파라미터 학습을 지원하는, 메모리 효율적인 옵티마이저입니다. 이 옵티마이저는 SGD와 유사한 메모리 효율성으로 AdamW 수준의 성능을 유지합니다. 극한의 메모리 효율성이 필요하다면 APOLLO의 rank 1 변형인 APOLLO-Mini를 사용할 수 있습니다. APOLLO 옵티마이저는 다음과 같은 특징을 지원합니다.
```
Comment (skwh54) [inline docs/source/ko/optimizers.md]:
```suggestion
[Approximated Gradient Scaling for Memory Efficient LLM Optimization (APOLLO)](https://github.com/zhuhanqing/APOLLO) 는 사전 학습과 미세 조정 모두에 대해 전체 파라미터 학습을 지원하는, 메모리 효율적인 옵티마이저입니다. 이 옵티마이저는 SGD와 유사한 메모리 효율성으로 AdamW 수준의 성능을 유지합니다. 극한의 메모리 효율성이 필요하다면 APOLLO의 rank 1 변형인 APOLLO-Mini를 사용할 수 있습니다. APOLLO 옵티마이저는 다음과 같은 특징을 지원합니다. 
```
Line 129에는 "fine-tuning"을 "미세 조정"으로 번역하셔서 해당 용어로 통일 제안 드립니다.

**Review by skwh54 — APPROVED** [review]

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🤗

**Comment by stevhliu** [comment]
build-doc

**Comment by stevhliu** [comment]
build-doc
